### ISSUE TITLE
v0.0.73 tarball causes a hash mismatch

### ISSUE THREAD

**Issue by michaeljohnclancy** [fort-nix/nix-bitcoin#500]
Hi,

Am getting this error when trying to run nix-shell with the latest tarball:

`error: hash mismatch in file downloaded from 'https://github.com/nixos/nixpkgs/archive/9a17f325397d137ac4d219ecbd5c7f15154422f4.tar.gz':
         specified: sha256:0vmkvi5x46q14dapiq134bcn8g1frsi9mcmi1ap9h47im1zlv2fq
         got:       sha256:1an6mzhrrawr696swhn5pk71k00xq5p4vp186rcfyj9f9zx27707
`

This is the nixpkgsUnstable hash in flake.lock. I can locally update the lockfile's nixpkgsUnstable entry to use the shell, but on deployment I would have to package up and host the updated tarball myself. I am relatively new to using nix etc, so please let me know if I am misunderstanding something!

Thanks,
Michael

**Comment by erikarvstedt** [comment]
The hash is correct on my side:
```
nix-prefetch-url --unpack https://github.com/nixos/nixpkgs/archive/9a17f325397d137ac4d219ecbd5c7f15154422f4.tar.gz
# => 0vmkvi5x46q14dapiq134bcn8g1frsi9mcmi1ap9h47im1zlv2fq
```
Try removing the fetcher cache and re-run nix-shell:
```
rm ~/.cache/nix/fetcher-cache*
```
Content hashes ensure data integry and should never be changed without close inspection.

**Comment by michaeljohnclancy** [comment]
I tried the above but still get the same issue!

After removing the cache, I also ran the same nix-prefetch-url command and get a different hash from you:
`1an6mzhrrawr696swhn5pk71k00xq5p4vp186rcfyj9f9zx27707`

When I run `nix flake lock --update-input nixpkgsUnstable` I am then able to run nix-shell, but I can see that its not exactly a solution.

For context, I am on MacOS and installed nix on this machine for the first time yesterday.

**Comment by erikarvstedt** [comment]
Do the hashes for the other flake inputs work normally?

**Comment by michaeljohnclancy** [comment]
Yes from what I can tell! After updating that input, and that input alone, I was able to run the nix-shell.
Version reported is `nix (Nix) 2.9.1`

**Comment by erikarvstedt** [comment]
Could you upload your version of nixpkgs that causes the hash mismatch?
```bash
path=$(nix flake metadata github:NixOS/nixpkgs/9a17f325397d137ac4d219ecbd5c7f15154422f4 | grep -o /nix/store.*)
 # Also, what's the value of path?
echo $path
curl --progress-bar -T <(tar cJ -C $path .) https://transfer.sh/nixpkgs | tee /dev/null
```

**Comment by michaeljohnclancy** [comment]
Thanks for this help btw! Just uploaded.

This is the value of path: `/nix/store/98qb8xfcjm4w0jn3if3h1bicwjshg4yx-source`

**Comment by erikarvstedt** [comment]
Can you post the transfer.sh URL that got output by curl?

**Comment by erikarvstedt** [comment]
Is the hash output of this command...
```
curl -s -L https://github.com/nixos/nixpkgs/archive/9a17f325397d137ac4d219ecbd5c7f15154422f4.tar.gz | sha256sum
```
... equal to`990c09d29a0adde8333fdb5cd57d9181dd2f269a47f99869da37b46b75fa461b`?
If not, please run the following:
```
curl --progress-bar -T <(curl -s -L https://github.com/nixos/nixpkgs/archive/9a17f325397d137ac4d219ecbd5c7f15154422f4.tar.gz) https://transfer.sh/nixpkgs2 | tee /dev/null
```

**Comment by michaeljohnclancy** [comment]
Yes it is!

**Comment by erikarvstedt** [comment]
Fixed in #501.
You can use this right away by setting the content of `nix-bitcoin-release.nix` to.
```
builtins.fetchTarball {
  url = "https://github.com/fort-nix/nix-bitcoin/archive/4579b44781fb283e7506f47e4588b1d24728b243.tar.gz";
  sha256 = "sha256:0d9fjxdj88qs9ghfk9xmgc248s45r1ic7m68kxr95gz8y4dk1b0f";
}
```

**Comment by michaeljohnclancy** [comment]
This works! Thanks very much - WIll I close issue or wait for merge into master?
